ultravox/model/data_processing.py

```diff
@@ -45,6 +49,9 @@ def _process(self, sample: datasets.VoiceSample) -> Dict[str, Any]:
             sample.messages, tokenize=False
         )
 
+        if self.include_alt_input:
+            alt_text = text.replace("<|audio|>", sample.audio_transcript or "")
```

Comment:
The `_get_sample()` method in `datasets.py` would be a more appropriate place to handle this logic. I think it’s acceptable to support empty audio transcripts in this context, as an audio file might simply contain silence or noise.

Reply:
My concern was more that if there was some logic error that led to a `None` transcript, we should probably handle that rather than silently inserting an empty transcript. Even if the audio is silent, there should still be a valid "" transcript.

Reply:
Agreed.

Reply:
I had to keep the `or ""` statement for now, otherwise some samples cause training failure. Will revisit this later as we start to clean up `datasets.py`